etcdserver/server.go

```diff
@@ -1686,15 +1725,21 @@ func (s *EtcdServer) mayRemoveMember(id types.ID) error {
 	}
 
 	// protect quorum if some members are down
-	m := s.cluster.Members()
-	active := numConnectedSince(s.r.transport, time.Now().Add(-HealthInterval), s.ID(), m)
-	if (active - 1) < 1+((len(m)-1)/2) {
+	votingMembers := s.cluster.VotingMembers()
+	numberOfVotingMembers := len(votingMembers)
+	numberOfActiveVotingMembers := numConnectedSince(s.r.transport, time.Now().Add(-HealthInterval), s.ID(), votingMembers)
+	memberToBeRemovedIsLearner := s.cluster.IsMemberExist(id) && s.cluster.Member(id).IsLearner
+	if !memberToBeRemovedIsLearner {
+		numberOfVotingMembers--
+		numberOfActiveVotingMembers--
+	}
+	if numberOfActiveVotingMembers < numberOfVotingMembers/2+1 {
```

Comment:
`quorum := numberOfVotingMembers/2+1`, just for additional clarity?

Reply:
Sounds good!